CONTRIBUTING.md

```diff
@@ -0,0 +1,193 @@
+
+#### 2.2 Levels of abstraction
+It is a best practice to abstract the code into different layers. For example, code related to communication with ROS should not include GUI changes. Common layers are GUI, Logic, Communication.
+
+
+### 3. Comments
+Comments can be a blessing or a cure. Proper documentation can help developers while bad documentation can obstruct or mislead them.
+
+#### 3.1 Useless comments
+Ideal code is self explanatory. Do not comment when it is obvious like
+
+```
+//Initialize ROS
+initRos();
+```
+
+This adds lines for the programmers to read without adding any value.
+
+#### 3.2 Rotted Comments
+One major issue with comments is that they rot with time. Before writing a comment, make sure that this comment will still be relevant after the code passes through a lot of changes.
+
+#### 3.3 Todos
+Do not write any `Todo` comments. These get easily lost and forgotten. Open an issue instead.
+
+#### 3.4 Commented code
+Avoid commenting code. A lot of the code that is commented, never gets uncommented again. It is a better practice to remove the code. We are using git as a version control tool, this means that old code can always be recuperated.
+ 
+
+### 4. Meaningful Names
+
+#### 4.1 Variable Names
+
+With the function that we improved above there is still a glaring issue :
+
+```
+void Commands::setTto(bool throttleTimeOut) {
+    String msg = "ASTRO Throttle Timeout " + String(throttleTimeOut ? "On" : "Off");
+    println(msg);
+
+}
+```
+
+What the heck is `TTO`? By observing this function and it's argument, it is obvious that it's for `Throttle Timeout`. However, imagine that you were stumbling upon a call to this function like this
+
+```
+//some code
+setTto(false);
+//more code
+```
+To know what `TTO` means, you need to go to the function definition which could be tedious if you have to do this often.
+
+Furthermore, you would expect `throttleTimeOut` to be a number, the timeout time, but it is a boolean.
+
+```
+void Commands::hasThrottleTimeout(bool hasThrottleTimeout) {
+    String msg = "ASTRO Throttle Timeout " + String(hasThrottleTimeout ? "On" : "Off");
+    println(msg);
+}
+```
+
+Boolean variable should start with `has` or `is`.
+
+
+```
+//some code
+hasThrottleTimeout(false);
+//more code
+```
+Now it is much cleaner when the call is being read.
+
+#### 4.2 Magic Values
+
+Magic values (or magic numbers) are values used with unexplained context.
+
+An example of the usage of a magic value :
+```
+if (isPressed(input[65]))
+```
+
+We do not know what `65` is.
+
+To fix this problem, we can simply defined a constant.
+
+```
+if (isPressed(input[A_KEY]))
+```
+
+
+Magic values should be avoided and replaced by variables or constants with names that self-describes their behavior.
+
+
+
+## Programming Specific Guidelines
+
+### Python
+
+We use the PEP8 Style Guide. Please refer to it [here](https://www.python.org/dev/peps/pep-0008/)
```

Comment:
Did you take [this](https://github.com/space-concordia-robotics/robotics-prototype/wiki/Code-Formatting-and-Conventions#formatting-guide-python) into consideration? We have this already in place for enforcing python programming style and standards using autoformatter/linter combo in `requirements-dev.txt`.

Reply:
there's a key in the yapf file that we can put `based_on_style = pep8`. Should we do it ? Maybe discuss IRL

Reply:
yeah might as well do it

Reply:
done